src/pages/expansions/StocktakeEditExpansion.js

```diff
@@ -0,0 +1,207 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2016
+ */
+
+import React from 'react';
+import { StyleSheet } from 'react-native';
+import PropTypes from 'prop-types';
+import autobind from 'react-autobind';
+import { Expansion } from 'react-native-data-table';
+import { dataTableStyles, expansionPageStyles } from '../../globalStyles';
+import { PageButton, ExpiryTextInput, PageInfo } from '../../widgets';
+import { GenericPage } from '../GenericPage';
+import { parsePositiveInteger } from '../../utilities';
+import { tableStrings, buttonStrings } from '../../localization';
+
+/**
+* Renders page to be displayed in StocktakeEditPage -> expansion.
+* @prop   {Realm}               database      App wide database.
+* @state  {Realm.Results}       items         the stocktakeItems of props.stocktake.
+*/
+export class StocktakeEditExpansion extends React.Component {
+  constructor(props) {
+    super(props);
+    this.item = props.data;
+    this.state = {};
+    autobind(this);
+  }
+
+  onEndEditing(key, stocktakeBatch, newValue) {
+    this.props.database.write(() => {
+      switch (key) {
+        case 'countedTotalQuantity': {
+          const newCountedQuantity = parsePositiveInteger(newValue);
+          if (newCountedQuantity === null) return;
+
+          stocktakeBatch.countedTotalQuantity = parsePositiveInteger(newValue);
+          // Should refresh StocktakeEditPage, because no changes are made to
+          // StocktakeItem object, but countedTotalQuantity getter result will be different
+          this.props.refreshParent();
```

Comment:
Wouldn't a "dataTypesLinked" on the parent genericPage make this redundant?

Reply:
yes indeed! good point.